### PR TITLE
Add lazy iterator for selectattr filter.

### PR DESCRIPTION
**Pull request by mattcoley**
WIP to lazily evaluate list filters.

### CODE REVIEW

**Comment by mattcoley** [comment]
@hs-lsong @boulter I still have to fix some code so the list test suite passes but thoughts on general implementation?

File: src/main/java/com/hubspot/jinjava/objects/collections/LazyList.java [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+  @Override
+  public boolean isEmpty() {
+    return !iterator.hasNext() && list.isEmpty();
+  }
+
+  @Override
+  public boolean contains(Object o) {
+
+    for (int i = 0; i < list.size(); i++) {
```
Comment (boulter) [inline src/main/java/com/hubspot/jinjava/objects/collections/LazyList.java]: hopefully this doesn't get called much 😢

File: src/main/java/com/hubspot/jinjava/objects/collections/LazyList.java [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    iterator.unpack();
+    return list.hashCode();
+  }
+
+  @Override
+  public boolean removeIf(Predicate<? super T> filter) {
+    iterator.unpack();
+    return list.removeIf(filter);
+  }
+}
```
Comment (boulter) [inline src/main/java/com/hubspot/jinjava/objects/collections/LazyList.java]: where did you pull this code from?

File: src/test/java/com/hubspot/jinjava/objects/collections/LazyListTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.google.common.collect.testing.features.ListFeature;
+
+@RunWith(AllTests.class)
+public class LazyListTest {
+
+  public static junit.framework.TestSuite suite() {
+    return ListTestSuiteBuilder.using(new TestStringListGenerator() {
+      @Override
+      protected List<String> create(String[] elements) {
+        // Fill here your collection with the given elements
```
Comment (boulter) [inline src/test/java/com/hubspot/jinjava/objects/collections/LazyListTest.java]: definitely would be good to test that it doesn't evaluate more than it needs to here.

**Comment by hs-lsong** [comment]
Cool, looks good.

**Pull request closed without merging**